src/Symfony/Component/Form/Console/Descriptor/Descriptor.php

```diff
@@ -128,9 +136,57 @@ protected function getOptionDefinition(OptionsResolver $optionsResolver, $option
             }
         }
 
+        if (isset($definition['deprecationMessage']) && \is_string($definition['deprecationMessage'])) {
+            $definition['deprecationMessage'] = strtr($definition['deprecationMessage'], array('%name%' => $option));
+        }
+
         return $definition;
     }
 
+    protected function filterTypesByDeprecated(array $types): array
```

Comment:
To be consistent with current code, should this rather live in the command and pass deprecated types & options to the descriptor as `$options`, so the descriptors' responsibility sticks to handling output?

Reply:
I agree, but in that case should we to move `collectOptions()`, `getParentOptionsResolver()`, etc, too?

Reply:
I guess it's okay as we aim to describe OptionResolver and ResolvedFormTypeInterface instances. What is a bit weird here to me is injecting the form registry.
But no strong opinion.

Reply:
Okey, done.